sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c

```diff
@@ -886,10 +885,10 @@ static void _receive(gnrc_pktsnip_t *pkt)
         gnrc_pktbuf_realloc_data(pkt, byteorder_ntohs(hdr->len));
     }
     else if (byteorder_ntohs(hdr->len) >
-             (gnrc_pkt_len_upto(pkt, GNRC_NETTYPE_IPV6) - sizeof(ipv6_hdr_t))) {
+             gnrc_pkt_len_upto(pkt, GNRC_NETTYPE_IPV6)) {
         DEBUG("ipv6: invalid payload length: %d, actual: %d, dropping packet\n",
               (int) byteorder_ntohs(hdr->len),
-              (int) (gnrc_pkt_len_upto(pkt, GNRC_NETTYPE_IPV6) - sizeof(ipv6_hdr_t)));
+              (int) gnrc_pkt_len_upto(pkt, GNRC_NETTYPE_IPV6));
```

Comment:
`gnrc_pkt_len_upto` returns actual packet size including IPv6 header. So we have to subtract `sizeof(ipv6_hdr_t)`.